### ISSUE TITLE
Wrong instance of HttpClient get's injected using typed clients with the same name but different namespace

### ISSUE THREAD

**Issue by JeroenHildering** [aspnet/Extensions#2369]
If you encounter a case where you're adding 2 typed HttpClients with the same class name using `AddHttpClient<IMyClient, MyClient>()`, but from different namespaces. The second instance of `HttpClient` gets injected into the first one.

### To Reproduce
1. Using version the latest version of .NET Core 2
2. Having this code in Startup.cs
```
services.AddHttpClient<NameSpaceX.IMyClient, NameSpaceX.MyClient>();
services.AddHttpClient<NameSpaceY.IMyClient, NameSpaceY.MyClient>();
```
3. Instance of `HttpClient` from typed client in `NameSpaceY` get's injected into typed client of `NameSpaceX`

### Expected behavior
Dependencies of `HttpClient` in typed clients in `NameSpaceY` and `NameSpaceX` are injected as you would expect.

### Additional context
In our case this specifically happened with a typed http client from a nuget package which happened to have exactly the same name. I suspect this happens due to the fact `fullName` is set to `false` here: https://github.com/aspnet/Extensions/blob/e580a9e1e86b52d55c8c5a084ecb3bd6106156d3/src/HttpClientFactory/Http/src/DependencyInjection/HttpClientFactoryServiceCollectionExtensions.cs#L211 

### Workaround
Rename either one of the class names of the typed clients.

**Comment by brandondahler** [comment]
Looks this is a dupe of #519.  

In version >= 3.0.0-preview6 an InvalidOperationException will be thrown: 
https://github.com/aspnet/Extensions/blob/17b28e83aa85612289e00fa250ec4647cf26bba5/src/HttpClientFactory/Http/src/DependencyInjection/HttpClientBuilderExtensions.cs#L545-L552

**Comment by rynowak** [comment]
This was fixed as part of the 3.0 release. The right solution if you have multiple types with the same type name is to give them logical names.

```
services.AddHttpClient(typeof(A.MyClient).FullName).AddTypedClient<A.MyClient>();
services.AddHttpClient(typeof(B.MyClient).FullName).AddTypedClient<B.MyClient>();
```
